### ISSUE TITLE
Bug with searching by multiple unique params

### ISSUE THREAD

**Issue by bryanmcgrane** [prisma/prisma-client-go#453]
Is searching by multiple unique params implemented? I have the following prisma schema:

`
model Tenant {
  id          String     @id @db.Uuid() @default(uuid())
  createdAt   DateTime   @default(now())
  updatedAt   DateTime   @updatedAt
  name        String     @unique @db.VarChar(127)
  devices     Device[]
}

model Device {
  id          String   @id @db.Uuid() @default(uuid())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  deviceId    String   @db.VarChar(127)
  tenant      Tenant   @relation(fields: [tenantId], references: [id])
  tenantId    String   @db.Uuid()
  @@unique([deviceId, tenantId])
}
`

And I'm trying to query by the @@unique param with:
`
deviceModel, err := client.Device.FindFirst(
  db.Device.DeviceID.Equals(device.DeviceID),
  db.Device.TenantID.Equals(tenant.ID)
).Exec(context.Background())
`

And I get an Error in query graph construction
`
pql error: Error in query graph construction: QueryParserError(QueryParserError { path: QueryPath { segments: ["Query", "findFirstDevice", "where", "DeviceWhereInput", "AND"] }, error_kind: InputUnionParseError { parsing_errors: [QueryParserError { path: QueryPath { segments: ["Query", "findFirstDevice", "where", "DeviceWhereInput", "AND"] }, error_kind: ValueTypeMismatchError { have: List([Object({"deviceId_tenantId": Object({"deviceId": String("de4cac0c-dcf9-41b6-886c-3c983f91009e"), "tenantId": String("fd067e27-389a-4c46-84d2-ae163d5e34f3")})})]), want: Object(DeviceWhereInput) } }, QueryParserError { path: QueryPath { segments: ["Query", "findFirstDevice", "where", "DeviceWhereInput", "AND", "DeviceWhereInput", "deviceId_tenantId"] }, error_kind: FieldNotFoundError }] } })
`

**Comment by steebchen** [comment]
I can not reproduce this. Can you please try again with the latest tag, and if that also doesn't work with the latest version in the master branch via `go get github.com/prisma/prisma-client-go@master`? Thanks.

**Comment by bryanmcgrane** [comment]
There was an error with my example above. Apologies on that. Here is the query that causes the issue:
```
deviceModel, err := dbClient.Device.FindFirst(
		db.Device.DeviceIDTenantID(
			db.Device.DeviceID.Equals(device.DeviceID),
			db.Device.TenantID.Equals(tenant.ID),
		),
	).Exec(context.Background())
```

**Comment by steebchen** [comment]
Got it, this one doesn't work indeed.

This currently only works with FindUnique. This is a bit hard to get right as the Prisma AST doesn't work super well with the current Go client syntax.

In the meanwhile, you should be able to use either FindUnique or FindFirst, and for the latter you can just specify the fields individually instead of explicitly with the index method – if you specify all fields the unique index has, it has the same performance anyway.

**Comment by bryanmcgrane** [comment]
Thank you for the clarification! I can work with that for now, but I would love to see this supported in the future.

**Comment by steebchen** [comment]
Can you share what you want to do specifically? Do you want to query by unique index + something else, or why doesn't FindUnique work for you?

**Comment by bryanmcgrane** [comment]
Definitely! I have a database that currently serves many customers. In order to facilitate this, I've added a Tenant FK column to every table in my DB. In order to protect against developers accidentally writing code that leaks data between tenants, I have created services that wrap requests to my models. I still want to leverage the power of Prisma's type system, but I need this layer of security. Here's an example of my DeviceService:

```
type DeviceService struct {
	db            *db.PrismaClient
}

func (s *DeviceService) Get(
	where db.DeviceEqualsUniqueWhereParam, 
	tenantParam db.TenantEqualsUniqueWhereParam) (*db.DeviceModel, error) {
	client := s.db

	deviceModel, err := client.Device.FindFirst(where, db.Device.Tenant.Where(tenantParam)).With(db.Device.Tenant.Fetch()).Exec(context.Background())
	if errors.Is(err, db.ErrNotFound) {
		log.Error(err)
		return nil, status.Errorf(codes.NotFound, "Device could not be found")
	} else if err != nil {
		log.Error(err)
		return nil, status.Errorf(codes.Internal, "Device could not be found")
	}

	return deviceModel, nil
}
```

I've encountered this problem many times in my career and have always solved it this way.

**Comment by steebchen** [comment]
Thanks for the info. I agree that it makes a lot of sense.

For now, you'll have to define both fields individually. In this case the `DeviceIDTenantID` would just be something like syntactic sugar, as you're not constrained to just that index, because in FindFirst every type of query is allowed.

Just to clarify, you should be able to query what you want by specifying the fields individually and it also shouldn't affect performance.
